cmd/cgctl/main.go

```diff
@@ -0,0 +1,141 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	v2 "github.com/containerd/cgroups/v2"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "cgctl"
+	app.Version = "1"
+	app.Usage = "cgroup v2 management tool"
+	app.Flags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "debug",
+			Usage: "enable debug output in the logs",
+		},
+		cli.StringFlag{
+			Name:  "mountpoint",
+			Usage: "cgroup mountpoint",
+			Value: "/sys/fs/cgroup",
+		},
+	}
+	app.Commands = []cli.Command{
+		newCommand,
+		delCommand,
+		listCommand,
+		statCommand,
+	}
+	app.Before = func(clix *cli.Context) error {
+		if clix.GlobalBool("debug") {
+			logrus.SetLevel(logrus.DebugLevel)
+		}
+		return nil
+	}
+	if err := app.Run(os.Args); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+var newCommand = cli.Command{
+	Name:  "new",
+	Usage: "create a new cgroup",
+	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name:  "enable",
+			Usage: "enable the controllers for the group",
+		},
+	},
+	Action: func(clix *cli.Context) error {
+		path := clix.Args().First()
+		c, err := v2.NewManager(clix.GlobalString("mountpoint"), path, nil)
+		if err != nil {
+			return err
+		}
+		if clix.Bool("enable") {
+			controllers, err := c.ListControllers()
+			if err != nil {
+				return err
+			}
+			if err := c.ToggleControllers(controllers, v2.Enable); err != nil {
```

Comment:
This needs to be done for the parent group?

Reply:
you don't need to enable this at all if the system is setup to enable them

Reply:
When you have `cpu` in the top group `/sys/fs/cgroup`, `cpu` is enabled for `/sys/fs/cgroup/foo`, but not enabled for `/sys/fs/cgroup/foo/bar`.

```console
root@suda-ws01:/sys/fs/cgroup# cat cgroup.controllers 
cpuset cpu io memory pids rdma
root@suda-ws01:/sys/fs/cgroup# mkdir foo
root@suda-ws01:/sys/fs/cgroup# cat foo/cgroup.controllers 
cpu io memory pids
root@suda-ws01:/sys/fs/cgroup# mkdir foo/bar
root@suda-ws01:/sys/fs/cgroup# cat foo/bar/cgroup.controllers 
root@suda-ws01:/sys/fs/cgroup#
```

Reply:
yes, its up to the creator of the cgroup to enable these.  I don't think the sub cgroup should modify the parent settings